src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -94,4 +95,22 @@ private function loadResources()
             unset($this->resourceFiles[$key]);
         }
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function warmUp($cacheDir)
+    {
+        if (null !== $this->options['cache_dir']) {
+            if (null !== $this->locale) {
+                $this->loadCatalogue($this->locale);
```

Comment:
Actually I rolled back because with last update in 2.7, it was broken. And I think it's weird to use file names to retrieve available locales.